### ISSUE TITLE
Request: allow method calls on constants

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop-ast#144]
Currently methods & predicates can be called on `self`:

```
(_ #predicate?(:arg, %1) ...)
(_ #method(:arg, %1) ...)
```

Request is to allow calling them on constants:

```
(_ #Foo.predicate?(:arg, %1) ...)
(_ #Foo::Bar.method(:arg, %1) ...)
(_ #::Global::X.method(:arg, %1) ...)
```

**Comment by pirj** [comment]
Just for the reference, the [idea](https://github.com/rubocop-hq/rubocop-rspec/pull/956#discussion_r483971890) was to allow usage of `#rspec.example_groups.regular` call chains, for namespacing purposes. Since this would only create complication in NP parser, and since Ruby has namespacing capabilities built-in, I completely support `RSpec::ExampleGroups.regular` syntax instead.

Just to make sure, is it correct that given:
```ruby
module A
  module B
    def self.a_method
       # ...
    end
  end
end

module RuboCop
  module Cop
    class SomeCop < Base
      include A
      def_node_matcher :a, '( #B.a_method ...)'
    end
  end
end
```
would be able to call `A::B.a_method`?

It would be a nice way to structure configurable DSL helpers.

**Comment by pirj** [comment]
If the namespace is relative, i.e. you don't have to specify `#A::B.a_method` when `A` is included, I'd go with constants, no method call chaining.

**Comment by marcandre** [comment]
Released as v1.1.0

**Comment by pirj** [comment]
@marcandre You're like a ⚡-man! 💯
This is just in time for the 2.0 RuboCop-RSpec release, thanks a lot!
